cpp/ql/src/experimental/Security/CWE/CWE-703/FindIncorrectlyUsedExceptions.ql

```diff
@@ -0,0 +1,38 @@
+/**
+ * @name Operator Find Incorrectly Used Exceptions
+ * @description --Finding places for the dangerous use of exceptions.
+ *              --For example, when the distructor throws an exception but does not handle it.
+ * @kind problem
+ * @id cpp/operator-find-incorrectly-used-exceptions
+ * @problem.severity warning
+ * @precision medium
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-703
+ *       external/cwe/cwe-248
+ *       external/cwe/cwe-390
+ */
+
+import cpp
+
+from FunctionCall fc, string msg
+where
+  exists(ThrowExpr texp |
+    texp.getEnclosingFunction() = fc.(DestructorCall).getTarget() and
+    not exists(TryStmt ts |
+      texp.getEnclosingStmt().getParentStmt*() = ts.getStmt() and
+      not ts.getACatchClause().isEmpty()
+    ) and
+    msg = "This distructor contains exeption no wrapped to try..catch blocks"
+  )
+  or
+  fc.getTarget() instanceof Constructor and
+  fc.getTargetType().(Class).getABaseClass+().hasGlobalOrStdName("exception") and
+  not fc.isInMacroExpansion() and
+  not exists(ThrowExpr texp | fc.getEnclosingStmt() = texp.getEnclosingStmt()) and
+  not exists(FunctionCall fctmp | fctmp.getAnArgument() = fc) and
+  not fc instanceof ConstructorDirectInit and
+  not fc.getEnclosingStmt() instanceof DeclStmt and
+  not fc instanceof ConstructorDelegationInit and
+  msg = "This object does not generate an exception."
```

Comment:
This case looks interesting.  Can you give an example (preferably a test case) demonstrating what it is trying to catch?

Reply:
Actually I am looking for objects of type `std :: exception` that do not throw exceptions.

a basic example can be seen in PR. I will try to make a test case a little later.

Reply:
So, broadly, it's looking for cases where a `std::exception` object is created but not as part of a `throw`?  e.g. the author writes
```
std::bad_alloc();
```
when they meant to write:
```
throw std::bad_alloc();
```